ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -16,36 +16,26 @@
 */
 package org.apache.spark.ml.h2o.algos
 
-import java.io._
-
 import hex.Model
 import hex.genmodel.utils.DistributionFamily
-import org.apache.hadoop.fs.Path
 import org.apache.spark.annotation.{DeveloperApi, Since}
 import org.apache.spark.h2o._
 import org.apache.spark.ml.h2o.param.{H2OAlgoParams, H2OModelParams}
 import org.apache.spark.ml.param.ParamMap
 import org.apache.spark.ml.util._
 import org.apache.spark.ml.{Estimator, Model => SparkModel}
 import org.apache.spark.sql.types.StructType
-import org.apache.spark.sql.{Dataset, SQLContext}
+import org.apache.spark.sql.{Dataset, SparkSession}
 import water.Key
 import water.support.H2OFrameSupport
 
 import scala.reflect.ClassTag
-
 /**
   * Base class for H2O algorithm wrapper as a Spark transformer.
   */
 abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M] : ClassTag]
-(parameters: Option[P])
```

Comment:
This was used just for internal purposes and we can do it nicely without specifying internal H2O params in constructor